xain_fl/coordinator/coordinator.py

```diff
@@ -393,8 +388,22 @@ def _handle_end_training_round(
         # submitting the updates and raise an exception if it is the wrong
         # round.
 
-        # record the request data
-        model_weights: ndarray = proto_to_ndarray(message.weights)
+        # HACK: Participants are currently not aware of they ID the
+        # coordinator uses to identify them, so they generate a UUID
+        # and use it to upload their results. The EndTrainingMessage
+        # contains that UUID, so that the coordinator can retrieve the
+        # weights.
+        #
+        # FIXME: handle the case where the participant didn't send a
+        # participant ID, or sent an invalid one. Reading from storage
+        # will fail in that case, and we hshould gracefully handle
+        # this by removing the participant.
+        fake_participant_id: str = message.participant_id
+        logger.info("downloading results", participant_id=participant_id)
+        model_weights: ndarray = self.local_weights_reader.read_weights(
+            fake_participant_id, self.current_round
+        )
+        logger.debug("done downloading results", participant_id=participant_id)
```

Comment:
can we make both logs about status (downloading/finished downloading) info or both debug to be consistent?

Reply:
done (as debug)

Reply:
Awesome! Thanks!